File: jeehyun/개별문제/SW Expert Academy/1220_Magnetic.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
	int T = 1;
	for (int test_case = 1; test_case <= T; test_case++) {
		int n, pole;
		cin >> n;

		vector<int> status(n);
		int answer = 0;

		for (int i = 0; i < n; i++) {
			for (int j = 0; j < n; j++) {
				cin >> pole;
				if (pole == 2 && status[j] == 1) {
					answer++;
					status[j] = 0;
				}
				else if (pole == 1)
					status[j] = 1;
			}
		}
		cout << "#" << test_case << " " << answer << "\n";
	}

	return 0;
}

File: jeehyun/개별문제/SW Expert Academy/1220_Magnetic.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Solution {
	private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	private static int map[][] = new int[100][100];
	
	public static void main(String args[]) throws Exception {
		StringBuilder sb = new StringBuilder();
		int T = 10;

        for(int test_case = 1; test_case <= T; test_case++) {
        	int n = Integer.parseInt(br.readLine());
        	
        	StringTokenizer st;
        	for (int i = 0; i < n; i++) {
        		st = new StringTokenizer(br.readLine(), " ");
        		for (int j = 0; j < n; j++) {
        			map[i][j] = Integer.parseInt(st.nextToken());
        		}
        	}
        	
        	int answer = 0;
        	for (int j = 0; j < n; j++) {
        		int spole = 0;
        		for (int i = 0; i < n; i++) {
        			if (map[i][j] == 2 && spole == 1) {
        				answer++;
        				spole = 0;
        			}
        			else if (map[i][j] == 1) spole = 1;
        		}
        	}        	
        	sb.append("#").append(test_case).append(" ").append(answer);
        	System.out.println(sb);
		}
	}
}